tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,199 @@
+                "No response received from the managed identity endpoint.";
+
+            IConfidentialClientApplication cca = CreateCCAWithProxy(s_baseURL);
+
+            //Act
+            try
+            {
+                AuthenticationResult authenticationResult = await cca
+                    .AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity()
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalServiceException ex)
+            {
+                result = ex.Message;
+            }
+
+            //Assert
+            Assert.AreSame(result, expectedClientException);
+        }
+
+        [DataTestMethod]
+        [DataRow(MsiAzureResource.WebApp, "", DisplayName = "System Identity Web App")]
+        [DataRow(MsiAzureResource.Function, "", DisplayName = "System Identity Function App")]
+        [DataRow(MsiAzureResource.WebApp, UserAssignedClientID, DisplayName = "User Identity Web App")]
+        [DataRow(MsiAzureResource.Function, UserAssignedClientID, DisplayName = "User Identity Function App")]
+        [DataRow(MsiAzureResource.WebApp, Mi_res_id, DisplayName = "ResourceID Web App")]
+        [DataRow(MsiAzureResource.Function, Mi_res_id, DisplayName = "ResourceID Function App")]
+        public async Task AcquireMSITokenAsync(MsiAzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            AuthenticationResult result = null;
+
+            using (new EnvVariableContext())
+            {
+                //Get the Environment Variables
+                Dictionary<string, string> envVariables = 
+                    await GetEnvironmentVariablesAsync(azureResource).ConfigureAwait(false);
+
+                //Set the Environment Variables
+                SetEnvironmentVariables(envVariables);
+
+                //form the http proxy URI 
+                string uri = s_baseURL + $"GetMSIToken?" +
+                    $"azureresource={azureResource.ToString().ToLowerInvariant()}&uri=";
+
+                //Create CCA with Proxy
+                IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+                //Act
+                try
+                {
+                    result = await cca
+                        .AcquireTokenForClient(s_msi_scopes)
+                        .WithManagedIdentity(userIdentity)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (MsalClientException ex)
+                {
+                    throw new Exception(ex.Message);
+                }
+            
+                //Assert
+                //1. MSI Helper service trims the access token, so that the access token from the MSI
+                //   is not fully returned to the calling service or test for security reasons
+                Assert.IsTrue(result.AccessToken.Length == 28);
```

Comment:
The length requirement seems a bit flaky. Maybe AT returned by the service should just be some constant - "MSI_token" or something?